### ISSUE TITLE
run locally . but doesn't run on Azure Devops Cloud pipeline

### ISSUE THREAD

**Issue by Farzad-Jalali** [MicrosoftDocs/azure-devops-docs#4445]
I done it, it runs locally just fine.
then I publish it and install in inside my organization account.
but when I try to run it in the cloud I get the below error :

Error: Cannot find module 'azure-pipelines-task-lib/task'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.&lt;anonymous&gt; (D:\a\_tasks\ScriptLoader_f1fd7724-d569-4740-b5f2-c0f32370c6bb\0.1.0\index.js:11:12)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
##[error]Exit code 1 returned from process: file name 'C:\agents\2.150.3\externals\node\bin\node.exe', arguments '"D:\a\_tasks\ScriptLoader_f1fd7724-d569-4740-b5f2-c0f32370c6bb\0.1.0\index.js"'.
##[section]Finishing: Echo lolo

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 44581f4a-93df-7507-3844-eadedf9a6cc4
* Version Independent ID: 33c7bcf0-1512-975f-5ef8-9f2a192a84ad
* Content: [Add a Build or Release Task](https://docs.microsoft.com/en-us/azure/devops/extend/develop/add-build-task?view=azure-devops#feedback)
* Content Source: [docs/extend/develop/add-build-task.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/extend/develop/add-build-task.md)
* Product: **devops**
* Technology: **devops-ecosystem**
* GitHub Login: @elbatk
* Microsoft Alias: **elbatk**

**Comment by thomps23** [comment]
Thanks for your question. It looks like you're working through an issue with your scenario or implementation, rather than an issue with the documentation. Here are a few options where you might consider asking your question:
- [Azure DevOps on Stack Overflow](https://stackoverflow.com/questions/tagged/azure-devops)
- [Azure DevOps Support Bot](https://azuredevopsvirtualagent.azurewebsites.net/)

If it turns out that this might be a bug in the service, you can [submit your bug]( https://developercommunity.visualstudio.com/content/problem/post.html?space=21) here.

**Comment by yanlinsun** [comment]
I ran into exactly the same issue. The root reason is that, if you check the task folder in the agent machine which runs your task, there is no 'node_modules' folder under the task folder. At first, I thought it was due to some errors in my extension; but, after hours of investigation, it is simply that the agent will not automatically install the required modules, it is expecting your task folder to include the node modules. The document does not mention any of this information.

So, to fix this, copy the 'node_modules' to 'buildAndReleaseTask'. But, as your task goes bigger, it is easy to exceed the size limit (50MB) of a vsix file. You'd better do an 'npm install --production' or 'npm prune --production' before copy the node folder.
Or write a script to build and pack everything.
